gpiozero/devices.py

```diff
@@ -441,3 +484,41 @@ def __repr__(self):
         except DeviceClosed:
             return "<gpiozero.%s object closed>" % self.__class__.__name__
 
+
+# Defined last to ensure Device is defined before attempting to load any pin
+# factory; pin factories want to load spi which in turn relies on devices (for
+# the soft-SPI implementation)
+def _default_pin_factory(name=os.getenv('GPIOZERO_PIN_FACTORY', None)):
+    group = 'gpiozero_pin_factories'
+    if name is None:
+        # If no factory is explicitly specified, try various names in
+        # "preferred" order. Note that in this case we only select from
+        # gpiozero distribution so without explicitly specifying a name (via
+        # the environment) it's impossible to auto-select a factory from
+        # outside the base distribution
+        #
+        # We prefer RPi.GPIO here as it supports PWM, and all Pi revisions. If
+        # no third-party libraries are available, however, we fall back to a
+        # pure Python implementation which supports platforms like PyPy
+        dist = pkg_resources.get_distribution('gpiozero')
+        for name in ('rpigpio', 'rpio', 'pigpio', 'native'):
+            try:
+                return pkg_resources.load_entry_point(dist, group, name)()
+            except Exception as e:
```

Comment:
If we're on a RPi3 with only the 'rpio' and 'pigpio' entry points available, would it select 'rpio' as the default factory (and then later fail when actually trying to use it, as rpio doesn't work on the Pi3), or is it clever enough to actually select 'pigpio' in that case?